pkg/registry/apis/datasource/connections.go

```diff
@@ -57,32 +58,38 @@ func (s *connectionAccess) ConvertToTable(ctx context.Context, object runtime.Ob
 }
 
 func (s *connectionAccess) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	ns := request.NamespaceValue(ctx)
 	ds, err := s.builder.getDataSource(ctx, name)
 	if err != nil {
 		return nil, err
 	}
-	return s.asConnection(ds), nil
+	return s.asConnection(ds, ns), nil
 }
 
 func (s *connectionAccess) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	ns := request.NamespaceValue(ctx)
+	if ns == "" {
+		// require a namespace so we do not need to support reverse mappings (yet)
```

Comment:
This means that:
http://localhost:3000/apis/testdata.datasource.grafana.app/v0alpha1/connections

fails, but:
http://localhost:3000/apis/testdata.datasource.grafana.app/v0alpha1/namespaces/default/connections

This simplifies things while we sort out multi-tenancy and is not important (yet) because we do not have any multi-tenant controllers